mlir/include/mlir/ExecutionEngine/CRunnerUtils.h

```diff
@@ -482,10 +482,16 @@ extern "C" MLIR_CRUNNERUTILS_EXPORT double rtclock();
 //===----------------------------------------------------------------------===//
 // Uses a seed to initialize a random generator and returns the generator.
 extern "C" MLIR_CRUNNERUTILS_EXPORT void *rtsrand(uint64_t s);
-// Returns a random number in the range of [0, m).
-extern "C" MLIR_CRUNNERUTILS_EXPORT uint64_t rtrand(void *, uint64_t m);
+// Uses a random number generator g and returns a random number
+// in the range of [0, m).
+extern "C" MLIR_CRUNNERUTILS_EXPORT uint64_t rtrand(void *g, uint64_t m);
 // Deletes the random number generator.
-extern "C" MLIR_CRUNNERUTILS_EXPORT void rtdrand(void *);
+extern "C" MLIR_CRUNNERUTILS_EXPORT void rtdrand(void *g);
+// Uses a random number generator g and std::shuffle to modify memref m
+// in place. m will be populated with unique and random numbers
+// in the range of [0, size of m).
```

Comment:
Using m above for [0,m) as scalar, and here as memref m is confusing. I addition using size m as the range adds to the confusion. So I would at least name the memref differently, like vref.

But also, rather than saying each number is in the range [0, size of vref) now, just say it will be a permutation of all numbers in that range, which seems more precise on what will happen?

Reply:
The reason why I chose "unique and random numbers" is I would like to emphasize the uniqueness of the numbers generated as it was the main reason why I couldn't use the existing rng (it doesn't guarantee unique numbers). But if you think permutation is more precise, I could change it to that.

Reply:
I think the only point I wanted to make is that if you say "each number is in the range [0, size of vref) " then [0,0,0,...] or any sequence with at least a few duplicates would be possible. By using permutation of the integers [0,sizeof vref) you make it more clear that every index precisely occurs once, but at a random place.